devtools/zenodo/zenodo_data_release.py

```diff
@@ -3,12 +3,14 @@
 import datetime
 import logging
 import os
-from collections.abc import Iterable
+import tempfile
+import time
 from dataclasses import dataclass
 from pathlib import Path
-from typing import BinaryIO
+from typing import IO
```

Comment:
See [this issue](https://github.com/python/typeshed/issues/7843) - properly type annotating "file-like-objects" in Python is apparently complicated, but the "blessed" option is `IO[bytes]`.